src/Microsoft.DotNet.Interactive.Tests/Parsing/SubmissionParserTests.cs

```diff
@@ -134,15 +134,24 @@ public void Directive_parsing_errors_are_available_as_diagnostics()
             .Should()
             .ContainSingle<DirectiveNode>()
             .Which;
-        node
-            .GetDiagnostics()
+
+        var diagnostics = node.GetDiagnostics();
+
+        diagnostics
+            .Should()
+            .ContainSingle(d => d.Severity == CodeAnalysis.DiagnosticSeverity.Error)
+            .Which
+            .LinePositionSpan.End.Character
+            .Should()
+            .Be(node.Span.End);
+
+        diagnostics
             .Should()
-            .ContainSingle(d => d.Severity == DiagnosticSeverity.Error)
+            .ContainSingle(d => d.Severity == CodeAnalysis.DiagnosticSeverity.Error)
             .Which
-            .Location
-            .SourceSpan
+            .LinePositionSpan.Start.Character
```

Comment:
Considering creating a local for the LinePositionSpan so that the code to select the diagnostic need not be repeated across the 2 assertions.